File: AMF5/src/test/java/javaPlatform/CustomPayloadValidationPluginTest.java
package javaPlatform;

import amf.apicontract.client.platform.AMFConfiguration;
import amf.apicontract.client.platform.OASConfiguration;
import amf.core.client.common.validation.ValidationMode;
import amf.core.client.platform.model.DataTypes;
import amf.core.client.platform.validation.AMFValidationReport;
import amf.core.client.platform.validation.payload.AMFShapePayloadValidator;
import amf.shapes.client.platform.model.domain.ScalarShape;
import org.junit.Test;

public class CustomPayloadValidationPluginTest {

    final String CUSTOM_MEDIATYPE = "application/my-cool-mediatype";

    @Test
    public void customPluginTest() {
        AMFConfiguration config = OASConfiguration.OAS30()
                .withShapePayloadPlugin(new CustomShapePayloadValidationPlugin(CUSTOM_MEDIATYPE));
        ScalarShape booleanOnlyShape = new ScalarShape().withDataType(DataTypes.Boolean());
        AMFShapePayloadValidator validator = config.elementClient().payloadValidatorFor(booleanOnlyShape, CUSTOM_MEDIATYPE, ValidationMode.StrictValidationMode());
        AMFValidationReport report = validator.syncValidate("somethingNotRelevant");
        assert(report.conforms());
    }
}

File: AMF5/src/test/scala/scalaPlatform/CustomPayloadValidationPluginTest.scala
package scalaPlatform

import amf.apicontract.client.scala.OASConfiguration
import amf.core.client.common.validation.StrictValidationMode
import amf.core.client.platform.model.DataTypes
import amf.shapes.client.scala.model.domain.ScalarShape
import org.scalatest.flatspec.AsyncFlatSpec
import org.scalatest.matchers.should.Matchers

class CustomPayloadValidationPluginTest extends AsyncFlatSpec with Matchers {

  val CUSTOM_MEDIATYPE = "application/my-cool-mediatype"

  "AMF" should "let the user plugin and use a custom payload validation plugin" in {

    val config = OASConfiguration.OAS30().withPlugin(new CustomShapePayloadPlugin(CUSTOM_MEDIATYPE))
    val booleanOnlyShape = ScalarShape().withDataType(DataTypes.Boolean);
    val validator = config.elementClient().payloadValidatorFor(booleanOnlyShape, CUSTOM_MEDIATYPE, StrictValidationMode);
    validator.validate("somethingNotRelevant").map { report =>
      report.conforms shouldBe true
      validator shouldBe a[MySpyShapeValidator]
      validator.asInstanceOf[MySpyShapeValidator].isExecuted shouldBe true
    }
  }
}
